File: mobile/lib/features/contest/domain/entities/contest.dart
import 'package:equatable/equatable.dart';

class Contest extends Equatable {
  final String id;
  final String title;
  final String description;
  final DateTime startsAt;
  final DateTime endsAt;
  final DateTime createdAt;
  final bool? live;
  final bool? hasRegistered;
  final int timeLeft;
  final int liveRegister;
  final int virtualRegister;

  const Contest({
    required this.id,
    required this.title,
    required this.description,
    required this.startsAt,
    required this.endsAt,
    required this.createdAt,
    required this.live,
    required this.hasRegistered,
    required this.timeLeft,
    required this.liveRegister,
    required this.virtualRegister,
  });

  @override
  List<Object?> get props => [
        id,
        title,
        description,
        startsAt,
        endsAt,
        createdAt,
        live,
        hasRegistered,
        timeLeft,
        liveRegister,
        virtualRegister,
      ];
}

File: new_backend/src/models/contest.ts
import { Document, Schema, model } from 'mongoose';

export interface IContestDocument extends Document {
  description: string;
  departmentId: Schema.Types.ObjectId;
  liveRegister: number,
  virtualRegister: number,
  title: string;
  startsAt: Date;
  endsAt: Date;
}

const ContestSchema: Schema<IContestDocument> = new Schema(
  {
    description: {
      type: String,
      default: "🚀 Ignite your academic journey with our exclusive High School Contest! 🏆 Dive into diverse subjects, enhance critical skills, and win exciting prizes!",
    },
    departmentId: {
      type: Schema.Types.ObjectId,
      ref: "Department",
      required: [true, "Department is required."]
    },
    title: {
      type: String,
      default: "SkillBridge Contest"
    },
    liveRegister: {
      type: Number,
      default: 0
    },
    virtualRegister: {
      type: Number,
      default: 0
    },
    startsAt: {
      type: Date,
      required: true,
    },
    endsAt: {
      type: Date,
      required: true,
      validate: {
        validator: function (endsAt: Date) {
          return endsAt > this.startsAt;
        },
        message: 'End date must be greater than start date.',
      },
    },
  },
  {
    timestamps: {
      createdAt: 'createdAt',
      updatedAt: 'updatedAt',
    },
  }
);

const Contest = model<IContestDocument>('Contest', ContestSchema);

export default Contest;
